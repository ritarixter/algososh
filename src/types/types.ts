import { ElementStates } from "./element-states";

export interface IArr {
  number: number;
  elState?: ElementStates;
}

export interface IArrChar {
  0?: string;
  elState?: ElementStates;
}

export interface IQueue {
  number: string;
  elState?: ElementStates;
}

export interface IList {
  number?: string;
  elState?: ElementStates;
  isProgressing?: Boolean;
}

export interface IButtonsStatusSort {
  newArray: {
    disabled: boolean;
  };
  ascendingSort: {
    disabled: boolean;
    loading: boolean;
  };
  descendingSort: {
    disabled: boolean;
    loading: boolean;
  };
}