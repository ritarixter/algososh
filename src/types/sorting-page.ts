import { ElementStates } from "./element-states";

export interface IArrSorting {
 number?: number;
 elState?: ElementStates;
}