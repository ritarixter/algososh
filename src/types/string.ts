import { ElementStates } from "./element-states";

export interface IArrChar {
 0?: string;
 elState?: ElementStates;
}