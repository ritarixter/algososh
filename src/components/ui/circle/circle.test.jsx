import React from "react";
import renderer from "react-test-renderer";
import { render, fireEvent, screen } from "@testing-library/react";
import { Circle } from "./circle";
import { ElementStates } from "../../../types/element-states";

describe("Элемент Circle работает корректно", ()=>{
  it('Circle без буквы рендерится корректно', ()=>{
    const tree = renderer.create(<Circle/>).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Circle с буквой рендерится корректно', ()=>{
    const tree = renderer.create(<Circle letter="Текст"/>).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Circle с head рендерится корректно', ()=>{
    const tree = renderer.create(<Circle head="5"/>).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Circle с react-элементом в head рендерится корректно', ()=>{
    const tree = renderer.create(<Circle head={<Circle/>}/>).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Circle с tail рендерится корректно', ()=>{
    const tree = renderer.create(<Circle tail="5"/>).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Circle с react-элементом в tail рендерится корректно', ()=>{
    const tree = renderer.create(<Circle tail={<Circle/>}/>).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Circle с index рендерится корректно', ()=>{
    const tree = renderer.create(<Circle index={2}/>).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Circle с isSmall ===  true рендерится корректно', ()=>{
    const tree = renderer.create(<Circle isSmall={true}/>).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Circle с состоянием default рендерится корректно', ()=>{
    const tree = renderer.create(<Circle state={ElementStates.Default}/>).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Circle с состоянием changing рендерится корректно', ()=>{
    const tree = renderer.create(<Circle state={ElementStates.Changing}/>).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Circle с состоянием modified рендерится корректно', ()=>{
    const tree = renderer.create(<Circle state={ElementStates.Modified}/>).toJSON();
    expect(tree).toMatchSnapshot();
  });
})