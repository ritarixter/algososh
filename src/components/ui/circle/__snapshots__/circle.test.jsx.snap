// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Элемент Circle работает корректно Circle без буквы рендерится корректно 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   default"
    data-testid="circle"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`Элемент Circle работает корректно Circle с head рендерится корректно 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head string"
  >
    5
  </div>
  <div
    className="circle   default"
    data-testid="circle"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`Элемент Circle работает корректно Circle с index рендерится корректно 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   default"
    data-testid="circle"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  >
    2
  </p>
  <div
    className="text text_type_input text_color_input mt-4 absolute tail60 element"
  />
</div>
`;

exports[`Элемент Circle работает корректно Circle с isSmall ===  true рендерится корректно 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle  small default"
    data-testid="circle"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`Элемент Circle работает корректно Circle с react-элементом в head рендерится корректно 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  >
    <div
      className="content "
    >
      <div
        className="text text_type_input text_color_input mb-4 absolute head element"
      />
      <div
        className="circle   default"
        data-testid="circle"
      >
        <p
          className="text text_type_circle text_color_input letter"
        />
      </div>
      <p
        className="text text_type_input text_color_input mt-4 absolute index"
      />
      <div
        className="text text_type_input text_color_input mt-4 absolute tail30 element"
      />
    </div>
  </div>
  <div
    className="circle   default"
    data-testid="circle"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`Элемент Circle работает корректно Circle с react-элементом в tail рендерится корректно 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   default"
    data-testid="circle"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  >
    <div
      className="content "
    >
      <div
        className="text text_type_input text_color_input mb-4 absolute head element"
      />
      <div
        className="circle   default"
        data-testid="circle"
      >
        <p
          className="text text_type_circle text_color_input letter"
        />
      </div>
      <p
        className="text text_type_input text_color_input mt-4 absolute index"
      />
      <div
        className="text text_type_input text_color_input mt-4 absolute tail30 element"
      />
    </div>
  </div>
</div>
`;

exports[`Элемент Circle работает корректно Circle с tail рендерится корректно 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   default"
    data-testid="circle"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 string"
  >
    5
  </div>
</div>
`;

exports[`Элемент Circle работает корректно Circle с буквой рендерится корректно 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   default"
    data-testid="circle"
  >
    <p
      className="text text_type_circle text_color_input letter"
    >
      Текст
    </p>
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`Элемент Circle работает корректно Circle с состоянием changing рендерится корректно 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   changing"
    data-testid="circle"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`Элемент Circle работает корректно Circle с состоянием default рендерится корректно 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   default"
    data-testid="circle"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`Элемент Circle работает корректно Circle с состоянием modified рендерится корректно 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   modified"
    data-testid="circle"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;
